native/src/main/java/io/ballerina/lib/data/xmldata/xml/XmlParser.java

```diff
@@ -343,11 +344,15 @@ private void addTextToCurrentNodeIfExpTypeIsArray(ArrayType fieldType, BString b
                     bText, xmlParserData);
             case TypeTags.ANYDATA_TAG, TypeTags.JSON_TAG -> {
                 BArray tempArr = (BArray) ((BMap<BString, Object>) xmlParserData.nodesStack.peek()).get(bFieldName);
-                Object nextValue = tempArr.get(tempArr.getLength() - 1);
-                if (!(nextValue instanceof BMap)) {
+                // TODO: Using this -2 here seems strange for me. Find a better way to handle this.
+                HashMap<String, Integer> indexes
```

Comment:
I think using `-2` is ok, but it is better we can have a check for the length. wdyt?

Reply:
Sure. I will update that

Reply:
Checking the length here seems wrong to me since it always have more than 2 elements. There is no proper error we can log for such a case.